src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -631,17 +631,20 @@ public void run()
          previous = now;
 
          logPoolState("Before cleanup\t");
-         final List<PoolBagEntry> bag = connectionBag.values(STATE_NOT_IN_USE);
-         int removable = bag.size() - config.getMinimumIdle();
-         for (PoolBagEntry bagEntry : bag) {
-            if (removable > 0 && idleTimeout > 0L && clockSource.elapsedMillis(bagEntry.lastAccess, now) > idleTimeout) {
-               if (connectionBag.reserve(bagEntry)) {
-                  closeConnection(bagEntry, "(connection passed idleTimeout)");
-                  removable--;
-               }
-            }
-         }
-         
+          if(idleTimeout > 0L) {
+              final List<PoolBagEntry> bag = connectionBag.values(STATE_NOT_IN_USE);
+              int removable = bag.size() - config.getMinimumIdle();
+              if (removable > 0) {
+                  for (PoolBagEntry bagEntry : bag) {
+                      if (clockSource.elapsedMillis(bagEntry.lastAccess, now) > idleTimeout) {
```

Comment:
please remove outer if (removable > 0) and add it here:
if (removable > 0 && clockSource.elapsedMillis(bagEntry.lastAccess, now) > idleTimeout)